app/code/Magento/ImportExport/Model/ResourceModel/Import/Data.php

```diff
@@ -58,20 +66,7 @@ protected function _construct()
      */
     public function getIterator()
     {
-        $connection = $this->getConnection();
-        $select = $connection->select()->from($this->getMainTable(), ['data'])->order('id ASC');
-        $stmt = $connection->query($select);
-
-        $stmt->setFetchMode(\Zend_Db::FETCH_NUM);
-        if ($stmt instanceof \IteratorAggregate) {
-            $iterator = $stmt->getIterator();
-        } else {
-            // Statement doesn't support iterating, so fetch all records and create iterator ourself
-            $rows = $stmt->fetchAll();
-            $iterator = new \ArrayIterator($rows);
-        }
-
-        return $iterator;
+        return $this->iteratorFactory->create();
```

Comment:
Iterator factory should produce `\Iterator` objects accodring to the method docblock